data/tools/check_mixed_indent

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+# encoding: utf-8
+"""
+Flag an error if a single file uses both tabs and spaces for indenting.
+
+This does not change files, on the assumption that the mess is better
+handled with a text editor or a code reformatting tool.
+"""
+
+import  argparse, codecs, re, sys
+
+def check_indent(file):
+    has_tabs = False
+    with codecs.open(file, "r", "utf8") as lines:
+        for line in lines:
+            if re.match("\t", line):
+                has_tabs = True
+                break
+
+        if has_tabs:
+            for line in lines:
+                if re.match("    ", line):
+                    print("{file} uses both tabs and spaces for indents".format(file=file))
+                    return False
+
+        # Only one type of indent appears at the start of each line, so this only needs to
+        # check one way round for mixtures on the same line.
+        if has_tabs:
+            mixed_line_re = re.compile("\t+ ")
+        else:
+            mixed_line_re = re.compile(" +\t")
+        for line in lines:
+            if mixed_line_re.match(line):
+                print("{file} indents with tabs and spaces on the same line".format(file=file))
+                return False
+
+    return True
+
+if __name__ == "__main__":
+    ap = argparse.ArgumentParser(usage=__doc__)
+    ap.add_argument("files", metavar="string", nargs="*", help="Read input from these files")
+    options = ap.parse_args()
+
+    all_files_ok = True
+
+    for file in options.files:
+        try:
+            all_files_ok &= check_indent(file)
```

Comment:
In this line there's `&=`, which is a bitwise operator.
It'd be better to use this:

    if not check_indent(file):
        all_files_ok = False